gqcp/include/QCModel/CI/LinearExpansion.hpp

```diff
@@ -559,6 +559,15 @@ class LinearExpansion {
     enable_if_t<std::is_same<Z, SeniorityZeroONVBasis>::value, TwoDM<double>> calculate2DM() const { return this->calculateSpinResolved2DM().spinSummed(); }
 
 
+    /**
+     *  Calculate the one-electron density matrix for a full spin-resolved wave function expansion.
+     * 
+     *  @return the total (spin-summed) 1-DM
+     */
+    template <typename Z = ONVBasis>
+    enable_if_t<std::is_same<Z, SeniorityZeroONVBasis>::value, OneDM<double>> calculateSpinDensity() const { return this->calculateSpinResolved1DM().spinDensity(); }
```

Comment:
I don't think this API belongs to `LinearExpansion`. I'm afraid we've hit the moment where we'll have to introduce our density matrix types to the Python bindings! Could you open up an issue for this?

Reply:
Yep, one issue coming up!

Reply:
Yes, you can keep them here.